### ISSUE TITLE
Audit executable help strings

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#2543]
EvolveBurgersStep (and any other Burgers executable) reads
```
Evolve the Burgers equation.

The analytic solution is: Linear
The numerical flux is:    LocalLaxFriedrichs
```
I assume the other systems have similar problems.

This information does not appear to be anywhere else in the program help text.  It can be printed by the option parser, but not until it tries to parse the relevant section (e.g., AnalyticSolution) of an input file.  (A valid input file has to name the solution and flux.)

We've standardized on having the solution in the executable name, so we can probably drop that without causing a problem.  Don't know about the numerical flux.

**Comment by nilsdeppe** [comment]
I agree, we shouldn't list the solution in the help text for the executable.

With the changes I'm making to boundary corrections we will be able to select the correction at runtime, which means it also shouldn't be in the help text of the executable at that point.
